test/integration/targets/async/tasks/main.yml

```diff
@@ -279,7 +279,18 @@
   - name: remove custom tmp dir after test
     file:
       path: '{{ custom_async_tmp }}'
-      state: absent
+
+- name: test async with $HOME variable expansion
+  command: echo "test"
+  async: 5
+  poll: 1
+  vars:
+    ansible_async_dir: "$HOME/.ansible_test_var_expansion"
+
+- name: verify $HOME was expanded and cleanup
+  file:
+    path: "{{ ansible_env.HOME }}/.ansible_test_var_expansion"
+    state: absent
```

Comment:
Can you also add a test using async_status to cleanup an async directory? https://docs.ansible.com/ansible/latest/collections/ansible/builtin/async_status_module.html#parameter-mode